Formula/c/capnp.rb

```diff
@@ -24,10 +24,25 @@ class Capnp < Formula
   depends_on "cmake" => :build
 
   def install
-    mkdir "build" do
-      system "cmake", "..", *std_cmake_args
-      system "make", "install"
-    end
+    # Build shared library
+    system "cmake", "-S", ".", "-B", "build_shared",
+                    "-DBUILD_SHARED_LIBS=ON",
+                    "-DCMAKE_POSITION_INDEPENDENT_CODE=ON",
+                    "-DCMAKE_INSTALL_RPATH=#{rpath}",
+                    *std_cmake_args
+    system "cmake", "--build", "build_shared"
+    system "cmake", "--install", "build_shared"
+
+    # Build static library
+    system "cmake", "-S", ".", "-B", "build_static",
+                    "-DBUILD_SHARED_LIBS=OFF",
+                    "-DCMAKE_POSITION_INDEPENDENT_CODE=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "build_static"
+
+    # override everything via static, will clobber executables
+    # and include but thats the desired behavior
```

Comment:
```suggestion
    # override everything via static, will clobber executables
    # and include but thats the desired behavior
```
I disagree; that kind of defeats the purpose of having shared libraries. Let's just link the executables to the shared libraries.

Reply:
sure, how would you get this command to work:
`    file.write shell_output("#{bin}/capnp id").chomp + ";\n"`